File: frontend/src/api/packing.ts
import { api } from './client'

export const packingApi = {
  submit: (body: {
    cartridge_type: string
    resin: string
    lot_number: string
    units_packed: number
    notes?: string
    as_operator?: string
  }) => api.post<{ ok: boolean; message: string }>('/packing/submit', body),
}

File: api/schemas/packing.py
from pydantic import BaseModel, Field


class PackingSubmitRequest(BaseModel):
    cartridge_type: str  # resolved code, e.g. "V2"
    resin: str
    lot_number: str
    units_packed: int = Field(ge=1)
    notes: str = ""
    as_operator: str | None = None
